be/src/agent/heartbeat_server.h

```diff
@@ -79,6 +79,9 @@ class HeartbeatServer : public HeartbeatServiceIf {
 
     StatusOr<CmpResult> compare_master_info(const TMasterInfo& master_info);
 
+    // New function to print TMasterInfo object as a string
+    std::string print_master_info(const TMasterInfo& master_info) const;
```

Comment:
this interface is not necessary to be part of the HeartbeatServer, for just because HeartbeatServer uses the interface to print the master info.

Try to find a better place for this interface, e.g. common/ or util/ ?